File: 3748-sort-matrix-by-diagonals/sort-matrix-by-diagonals.py
class Solution(object):
    def sortMatrix(self, grid):
        o=[];
        n=len(grid);
        for i in range(len(grid)-1,0,-1):
            t=i;
            j=0;
            temp=[];
            while(t<len(grid)):
                temp.append(grid[j][t]);
                j+=1;
                t+=1;
            temp.sort();
            o.append(temp);
        
        for i in range(len(grid)):
            t=i;
            j=0;
            temp=[];
            while(t<len(grid)):
                temp.append(grid[t][j]);
                j+=1;
                t+=1;
            temp.sort(reverse=True);
            o.append(temp);
        arr=[];
        for i in o:
            for j in i:
                arr.append(j);
        ind=0;
        for i in range(len(grid)-1,0,-1):
            t=i;
            j=0;
            temp=[];
            while(t<len(grid)):
                grid[j][t]=arr[ind]
                j+=1;
                t+=1;
                ind+=1;
        for i in range(len(grid)):
            t=i;
            j=0;
            temp=[];
            while(t<len(grid)):
                grid[t][j]=arr[ind]
                j+=1;
                t+=1;
                ind+=1;
        return grid;
        """
        :type grid: List[List[int]]
        :rtype: List[List[int]]
        """
        

File: 3748-sort-matrix-by-diagonals/sort-matrix-by-diagonals.java
class Solution {
    public int[][] sortMatrix(int[][] grid) {
        int n=grid.length;
        ArrayList<Integer> temp=new ArrayList<Integer>();
        for(int i=n-1;i>=0;i--){
            int t=i;
            int j=0;
            ArrayList<Integer> arr=new ArrayList<Integer>();
            while(t<n){
                arr.add(grid[j][t]);
                t+=1;
                j+=1;

            }
            Collections.sort(arr);
            temp.addAll(arr);
        }
        for(int i=0;i<n;i++){
            int t=i;
            int j=0;
            ArrayList<Integer> arr=new ArrayList<Integer>();
            while(t<n){
                arr.add(grid[t][j]);
                t+=1;
                j+=1;
            }
            Collections.sort(arr,Collections.reverseOrder());
            temp.addAll(arr);
        }
        int ind=0;
        for(int i=n-1;i>=0;i--){
            int t=i;
            int j=0;
            while(t<n){
                grid[j][t]=temp.get(ind++);
                t+=1;
                j+=1;
            }
        }
        for(int i=0;i<n;i++){
            int t=i;
            int j=0;
            while(t<n){
                grid[t][j]=temp.get(ind++);
                t+=1;
                j+=1;
            }
        }
        return grid;        
    }
}